text/3579-pki.md

```diff
@@ -0,0 +1,262 @@
+
+Authentication to the CloudHSM instance shall be limited to only a [Signing Console][signing-console] at the network layer; at the application layer, we shall create IAM roles and accounts for all members of the quorum. These accounts will required two factor authentication and have restrictive password, timeout, and security policies applied. These roles shall only be used for Quorum signing events. There will be documentation for the quorum members on best practices for handling their credentials.
+
+These individuals should be available in the event quorum is needed for root key operations. These roles can and should be reappointed as needed, but to simplify logistics, these roles should not require rotation more often than 2-3 years. (Operations requiring quorum are expected to be rare, and an annual rotation would make rotation of the quorum group the vast majority of quorum operations.)
+
+##### When the Quorum will be needed
+[when-the-quorum-will-be-needed]: #when-the-quorum-will-be-needed
+
+- Changes to quorum membership or thresholds shall require a quorum (enforced by the HSM authentication)
+- Replacement of the Root Key in the event of expiration or compromise
+- Creation, signing, and issuance of a new Delegate Key
+- Creation, signing, and issuance of a replacement Delegate Key in the event of expiration or compromise
+
+##### Quorum Threat Mitigations
+
+- To reduce the feasibility of coercion or collusion, we recommend that no more than 3 (M-2) of the 9 people be affiliated with any one company or institution, or be residents or citizens of the same country. (If in the future the quorum threshold is changed, this limit should be changed accordingly to remain M-2).
+- If in the future a desire arises to modify the quorum, we strongly recommend growing the size of membership rather than reducing the quorum threshold. The quorum exists to mitigate against bad actors, and is not implemented to prove a majority.
+- To reduce the feasibility of coercion or collusion, we recommend that members of the quorum be selected from within the Rust project, community, Foundation staff and/or Foundation Board. This allows us to spread trust and responsibility among socially and fiduciarily responsible parties.
+- To prevent the destruction of the quorum, we recommend that no real world event occur in which 5 or more members of the quorum are present; we recommend at least 5 (M-2) members of the quorum remain at geographically diverse locations. (If in the future the quorum threshold is changed, this limit should be changed accordingly to remain M-2).
```

Comment:
This is probably not workable, there are going to be events where the presence of most of the quorum will be required. One is happening in a month.

Reply:
@pietroalbini The quorum members are not expected to all be project leaders, just trusted people. If we are having trouble ensuring that the majority of the quorum isn't at the same event, then we need to pick a broader set of quorum members. We could also, potentially, increase the total number of quorum members to more than 9, which would make it easier to not all be in the same place (but harder to establish a group with enough members in the first place).

At the end of the day, this is a *recommendation*, not a hard requirement, and it's always possible to recover from such an event with a large noisy key rotation, it's just not ideal.

Reply:
In the end trusted people (should) be project- or foundation-adjacent, and I can see cases where a project-wide meeting adjacent to a conference could result in breaking the quorum. Maybe the text could be softened a bit to prevent a possible hesitation about project-wide meetings?

Reply:
Addressed in the changes we just pushed by tempering this advice, and focusing more on geographically and politically distributed *normal residence* locations.